### DISCUSSION TITLE
Is it possible to mute a stream?

### DISCUSSION THREAD

**Discussion by brauliolomeli** [shogo4405/HaishinKit.dart#20] [Q&A] [answered]
I need to be able to toggle the audio on a stream, but I haven't been able to accomplish that.
I tried these once the stream has started:

1. `stream.attachAudio(AudioSource());` - kills the stream
2. `stream.audioSettings.muted=true` - does nothing
3. `stream.audioSettings = AudioSettings(muted: true) - does nothing

Tried all of them on an Android device.

**Comment by shogo4405** [comment, marked as answer]
I  added this option to Android project. https://github.com/shogo4405/HaishinKit.kt/pull/104/files
However, I forgot to add it to Flutter, so I don't provide the function now.

**Reply by OrbysSocialMediaLLC** [reply]
There is a workwround for this in flutter:
      setState(() {       
         _stream?.setHasAudio(false); 
  
      });

Just call it from a mute button and change the state back to tru to enable it again if needed. I hope this helps.

**Comment by VladProtko** [comment]
**HELP!**
Tell me how to implement this in IOS?
